src-test/Main.hs

```diff
@@ -159,14 +166,44 @@ cmdYaml2Event = do
       hPutStrLn stdout (ev2str True event)
       hFlush stdout
 
-
 cmdYaml2Event0 :: IO ()
 cmdYaml2Event0 = do
     inYamlDat <- BS.L.getContents
     print (length (parseEvents' inYamlDat))
   where
     parseEvents' = map (either (\(ofs,msg) -> error ("parsing error near byte offset " ++ show ofs ++ " (" ++ msg ++ ")")) id) . parseEvents
 
+cmdCheck :: IO() 
+cmdCheck = do
+  inYamlDat <- BS.L.getContents
+  let oldevstrm = sequence $ parseEvents inYamlDat
+  case (oldevstrm) of
+    Left (ofs,msg) -> do
+      case msg of
+        "" -> hPutStrLn stderr ("Parsing error near byte offset " ++ show ofs)
+        _  -> hPutStrLn stderr ("Parsing error near byte offset " ++ show ofs ++ " (" ++ msg ++ ")")
+      exitFailure
+    Right oldEvents -> do
+      let oldEvLen = length oldEvents
+          output = (writeEvents YT.UTF8 oldEvents)
```

Comment:
redundant brackets; oldEvLen can be inlined

Reply:
oldEvLen used later also